assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json

```diff
@@ -0,0 +1,18 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "prefix": "arm-automanage-custom-assignment",
+        "description": "Create a Custom Automanage Configuration Profile Assignment"
+    },
+    "resources": [
+       {
+            "type": "Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments",
+            "apiVersion": "2022-05-04",
+            "name": "[concat('${1:vmName}', '/Microsoft.Automanage/default')]",
+            "properties": {
+                "configurationProfile": "/subscriptions/${2:subscriptionId}/resourceGroups/${3:resourceGroup}/providers/Microsoft.Automanage/configurationProfiles/${4:profileName}"
```

Comment:
Please use the resourceId function instead.  See https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-test-cases#use-resource-id-functions

Reply:
Does this look right? 

```json
"configurationProfile": "[resourceId('${2:subscriptionId}', '${3:resourceGroup}', 'Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments', '${1:vmName}/Microsoft.Automanage/default')]"
```

Reply:
Can you test it with a deployment?  thx.

Reply:
alright ive made the update and verified that it works.